### ISSUE TITLE
isHexString in src/auth/index.js is inverted

### ISSUE THREAD

**Issue by chris-armstrong** [Azure/azure-mobile-apps-node#437]
isHexString() (below) is used to determine if the authorisation secret has been specified as a hex string or buffer:

```
function isHexString(source) {
    return source.match(/[^0-9a-fA-F]/g) !== null;
}
```

However, the logic is wrong - it returns false when the string IS a hex a string. This is causing issues for us in 2.1.6.

**Comment by danderson00** [comment]
Ouch. I rushed this out, my unit test was calling auth.decode instead of validate and I didn't make it fail first. Will get a fix out this afternoon.

**Comment by danderson00** [comment]
Resolved by https://github.com/Azure/azure-mobile-apps-node/commit/580c79ae74ce5d3838b6d6d1f1d1c31f758ab1a4

@chris-armstrong This has been released - 2.1.7. Sorry about that!
